src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlEnclaveSession.cs

```diff
@@ -56,4 +56,18 @@ internal SqlEnclaveSession(byte[] sessionKey, long sessionId/*, long counter*/)
             SessionId = sessionId;
         }
     }
+
+    internal class EnclaveSessionParameters
```

Comment:
Isn't is good this class to be immutable?

Reply:
We could use struct instead of a class here.

Reply:
This class wraps several strings that are required by the enclave session. From the changes in this PR, we only assign the values when we create the EnclaveSessionParameters object.